### ISSUE TITLE
[2015.8] pkg breaks for yum pkgs.latest if the packages has an epoch

### ISSUE THREAD

**Issue by gtmanfred** [saltstack/salt#31014]
It looks like it will install the package, but when salt goes back and makes sure that the latest version is installed, it does not take into account that a version may have an epoch that is not displayed by yum for some reason.

The quickest example is for libguestfs-tools, which has an epoch of 1.

```
----------
          ID: nova packages
    Function: pkg.latest
      Result: False
     Comment: The following packages failed to update: libguestfs-tools The following packages were already up-to-date: openstack-nova-compute, sysfsutils
     Started: 19:08:39.658538
    Duration: 7961.173 ms
     Changes:
              ----------
              libguestfs-tools:
                  ----------
                  new:
                      1.28.1-1.55.el7.centos
                  old:
```

[root@novacompute01 ~]# yum list installed | grep libguestfs-tools
libguestfs-tools.noarch              1:1.28.1-1.55.el7.centos @base
libguestfs-tools-c.x86_64            1:1.28.1-1.55.el7.centos @base

So the versions do not match, because the version displayed when installing is without the epoch.

https://github.com/saltstack/salt/blob/develop/salt/states/pkg.py#L1598

**Comment by gtmanfred** [comment]
@terminalmage ping :)

**Comment by cachedout** [comment]
Fixed via #31031
